cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java

```diff
@@ -117,6 +117,52 @@ public GoalOptimizer(KafkaCruiseControlConfig config,
     _proposalComputationTimer = dropwizardMetricRegistry.timer(MetricRegistry.name("GoalOptimizer", "proposal-computation-timer"));
   }
 
+  /**
+   * Generate unique goal priority for each of proposal precomputing thread.
+   */
+  private void populateGoalByPriorityForPrecomputing() {
+    List<List<Goal>> shuffledGoalList = new ArrayList<>();
+    // Guarantee one thread is working on the original goal priority.
+    shuffledGoalList.add(new ArrayList<>(_goalsByPriority.values()));
+
+    Integer goalCount = _goalsByPriority.values().size();
+    Integer goalIndex = goalCount - 2;
```

Comment:
Why do we subtract `2`? Let's add a comment to clarify that.

Reply:
added comment.